src/NumericalAlgorithms/DiscontinuousGalerkin/Actions/ImposeBoundaryConditions.hpp

```diff
@@ -0,0 +1,234 @@
+          }
+        },
+        make_not_null(&box));
+
+    for (const auto& direction : element.external_boundaries()) {
+      const auto& boundary_mesh = map_of_boundary_meshes.at(direction);
+      const auto packaged_data = db::apply<tmpl::transform<
+          package_arguments,
+          tmpl::bind<Tags::Interface, Tags::BoundaryDirections<volume_dim>,
+                     tmpl::_1>>>(
+          [&boundary_mesh, &direction,
+           &normal_dot_numerical_flux_computer ](const auto&... args) noexcept {
+            typename flux_comm_types::PackagedData ret(
+                boundary_mesh.number_of_grid_points(), 0.0);
+            normal_dot_numerical_flux_computer.package_data(
+                make_not_null(&ret), args.at(direction)...);
+            return ret;
+          },
+          box);
+
+      const auto external_packaged_data = db::apply<tmpl::transform<
+          package_arguments,
+          tmpl::bind<Tags::Interface,
+                     Tags::ExternalBoundaryDirections<volume_dim>, tmpl::_1>>>(
+          [&boundary_mesh, &direction,
+           &normal_dot_numerical_flux_computer ](const auto&... args) noexcept {
+            typename flux_comm_types::PackagedData ret(
+                boundary_mesh.number_of_grid_points(), 0.0);
+            normal_dot_numerical_flux_computer.package_data(
+                make_not_null(&ret), args.at(direction)...);
+            return ret;
+          },
+          box);
+
+      typename flux_comm_types::LocalData interior_data{};
+      interior_data.magnitude_of_face_normal = db::get<Tags::Interface<
+          Tags::BoundaryDirections<volume_dim>,
+          Tags::Magnitude<Tags::UnnormalizedFaceNormal<volume_dim>>>>(box)
+                                                   .at(direction);
+
+      interior_data.mortar_data.initialize(
+          boundary_mesh.number_of_grid_points());
+      interior_data.mortar_data.assign_subset(packaged_data);
+      // Need to make a fake mortar id for the non existent neighbor.
+      const auto mortar_id = std::make_pair(direction, element.id());
+
+      typename flux_comm_types::LocalData exterior_data{};
+      exterior_data.magnitude_of_face_normal = db::get<Tags::Interface<
+          Tags::ExternalBoundaryDirections<volume_dim>,
+          Tags::Magnitude<Tags::UnnormalizedFaceNormal<volume_dim>>>>(box)
+                                                   .at(direction);
+
+      exterior_data.mortar_data.initialize(
+          boundary_mesh.number_of_grid_points());
+      exterior_data.mortar_data.assign_subset(packaged_data);
+
+      if (tmpl::size<
+              typename flux_comm_types::LocalMortarData::tags_list>::value !=
+          tmpl::size<
+              typename flux_comm_types::PackagedData::tags_list>::value) {
+        // The local fluxes were not (all) included in the packaged
+        // data, so we need to add them to the mortar data
+        // explicitly.
```

Comment:
A lot of this is taken from the flux communication actions.  Can it be factored out?